### ISSUE TITLE
How to get one page object from another?

### ISSUE THREAD

**Issue by vaha** [icnocop/cuite#106]
Hi, is it possible to instantiate somehow one page object from another (on the "same levels" of UI control tree)?

Here is the example structure:
```
<div id="A">
  <div id="B"></div>
  <div id="C"></div>
</div>
```

So I have a Page class with the method that returns me a page object for the div B. How can I get page object for the div C from it?

To clarify the situation lets suppose that divB is a button and divC is a related confirmation dialog - that's why I want to get page object for the divC only from the particular page object and not from the class that describes the whole page.

**Comment by FantasticFiasco** [comment]
This seems to be more a problem of _search limit containers_ than page objects. There doesn't seem to be anything stopping you from defining two page objects, one representing `B` and one representing `C`, but with a shared search limit container of `A`. This of course means that the selectors, the way you find your controls, should take into account that both `B` and `C` will be searched when looking for a control.

It also means that when `B` is creating the `C` page object, it should do so with the same search limit container as it itself is configured to use.

Does any of this make any sense?

**Comment by vaha** [comment]
FantasticFiasco, thank you for the response. Agreed that my question mostly relates to the search limit container, I've mentioned just to describe situation better.

And one more small question - could we update somehow search limit container for page object? Or search limit container could be specified only using GetPageObject method?

**Comment by FantasticFiasco** [comment]
You are correct that there doesn't seem to be a way to update the search limit container when using the `GetPageObject<T>` method, the container of the current page object is automatically transferred to the new page object.

We can certainly create an overload that takes the container, would you be up to the task of creating a pull request for it?
